### PR TITLE
[FLINK-6711] Activate strict checkstyle for flink-connectors

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-connectors/flink-connector-nifi/src/main/java/org/apache/flink/streaming/connectors/nifi/NiFiDataPacket.java [hunk 1]
```diff
@@ -14,13 +14,14 @@
...
+
 package org.apache.flink.streaming.connectors.nifi;
 
 import java.util.Map;
 
 /**
- * <p>
- * The NiFiDataPacket provides a packaging around a NiFi FlowFile. It wraps both
+ *
+ * <p>The NiFiDataPacket provides a packaging around a NiFi FlowFile. It wraps both
```
Comment (greghogan) [inline flink-connectors/flink-connector-nifi/src/main/java/org/apache/flink/streaming/connectors/nifi/NiFiDataPacket.java]: Remove opening and closing paragraph tags and blank line.

File: flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public RMQSource(RMQConnectionConfig rmqConnectionConfig, String queueName,
 	/**
 	 * Creates a new RabbitMQ source. For exactly-once, you must set the correlation ids of messages
 	 * at the producer. The correlation id must be unique. Otherwise the behavior of the source is
-	 * undefined. In doubt, set {@param usesCorrelationId} to false. When correlation ids are not
+	 * undefined. In doubt, set usesCorrelationId to false. When correlation ids are not
```
Comment (greghogan) [inline flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java]: "If in doubt"?

File: flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/split/GenericParameterValuesProvider.java [hunk 3]
```diff
@@ -15,22 +15,22 @@
...
-/** 
- * 
+import java.io.Serializable;
+
+/**
  * This splits generator actually does nothing but wrapping the query parameters
  * computed by the user before creating the {@link JDBCInputFormat} instance.
- * 
+ *
  * */
```
Comment (greghogan) [inline flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/split/GenericParameterValuesProvider.java]: Empty line and extra asterisk.

File: flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java [hunk 4]
```diff
@@ -17,27 +17,27 @@
...
 /**
- * 
  * This is an example how to write streams into HBase. In this example the
  * stream will be written into a local Hbase but it is possible to adapt this
  * example for an HBase running in a cloud. You need a running local HBase with a
  * table "flinkExample" and a column "entry". If your HBase configuration does
  * not fit the hbase-site.xml in the resource folder then you gave to delete temporary this
  * hbase-site.xml to execute the example properly.
- * 
+ *
```
Comment (greghogan) [inline flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteStreamExample.java]: Empty line.

File: flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java [hunk 5]
```diff
@@ -80,11 +79,21 @@
...
 		CONSTANT,
 		EXPONENTIAL
 	}
 
+	/**
+	 * Provides a backoff policy for bulk requests. Whenever a bulk request is rejected due to resource constraints
+	 * (i.e. the client's internal thread pool is full), the backoff policy decides how long the bulk processor will
+	 * wait before the operation is retried internally.
+	 *
+	 * <p>This ia a proxy for version specific backoff policies.
```
Comment (greghogan) [inline flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java]: "ia" => "is"

File: flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer010.java [hunk 6]
```diff
@@ -62,11 +62,10 @@
 
 	private static final long serialVersionUID = 2324564345203409112L;
 
-
 	// ------------------------------------------------------------------------
 
 	/**
-	 * Creates a new Kafka streaming source consumer for Kafka 0.10.x
+	 * Creates a new Kafka streaming source consumer for Kafka 0.10.x .
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer010.java]: Extra space before period.

File: flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer08.java [hunk 7]
```diff
@@ -102,21 +99,21 @@
...
 
-	/** The properties to parametrize the Kafka consumer and ZooKeeper client */ 
+	/** The properties to parametrize the Kafka consumer and ZooKeeper client. */
 	private final Properties kafkaProperties;
 
 	// ------------------------------------------------------------------------
 
 	/**
-	 * Creates a new Kafka streaming source consumer for Kafka 0.8.x
+	 * Creates a new Kafka streaming source consumer for Kafka 0.8.x .
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer08.java]: Also has an extra space.

File: flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java [hunk 8]
```diff
@@ -15,23 +15,24 @@
...
 
 /**
  * Kafka 0.8 {@link KafkaTableSink} that serializes data in JSON format.
  */
 public class Kafka08JsonTableSink extends KafkaJsonTableSink {
-	
+
 	/**
-	 * Creates {@link KafkaTableSink} for Kafka 0.8
+	 * Creates {@link KafkaTableSink} for Kafka 0.8 .
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java]: Extra space before period.

File: flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java [hunk 9]
```diff
@@ -42,7 +43,7 @@ public Kafka08JsonTableSink(String topic, Properties properties, FlinkKafkaParti
 	}
 
 	/**
-	 * Creates {@link KafkaTableSink} for Kafka 0.8
+	 * Creates {@link KafkaTableSink} for Kafka 0.8 .
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java]: Extra space before period.

File: flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java [hunk 10]
```diff
@@ -47,10 +53,14 @@
 		BasicTypeInfo.LONG_TYPE_INFO };
 	private static final Properties PROPERTIES = createSourceProperties();
 
-	// Avro record that matches above schema
+	/**
+	 * Avro record that matches above schema.
+	 */
 	public static class AvroSpecificRecord extends SpecificRecordBase {
 
+		//CHECKSTYLE.OFF: ConstantNameCheck|
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java]: Is this intended to have a trailing pipe?

File: flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/testutils/FailingIdentityMapper.java [hunk 11]
```diff
@@ -22,30 +22,34 @@
...
 import org.slf4j.LoggerFactory;
 
 import java.util.Collections;
 import java.util.List;
 
-
-public class FailingIdentityMapper<T> extends RichMapFunction<T,T> implements
+/**
+ * A {@link RichMapFunction} that fails after the configured number of records have been processed.
+ * @param <T>
```
Comment (greghogan) [inline flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/testutils/FailingIdentityMapper.java]: Add extra line.

File: flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java (hunk 10 above)
Comment (zentol) [inline flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java]: For some reason checkstyle kept failing without the pipe; i have no idea why.

File: flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java (hunk 9 above)
Comment (zentol) [inline flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java]: This is somewhat intended, to clarify that it is the end of the sentence and not an incomplete version. Don't mind removing the space though.

Comment (greghogan) [inline flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka08JsonTableSink.java]: Okay, it was clearly a pattern but wasn't sure if it was an unexpected programmatic replacement. I think it's find leaving as is but expect that someone will eventually flag it for fixing.

File: flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/RollingSink.java [hunk 12]
```diff
@@ -266,7 +265,7 @@
 
 	/**
 	 * We use reflection to get the .truncate() method, this is only available starting with
-	 * Hadoop 2.7
+	 * Hadoop 2.7 .
```
Comment (greghogan) [inline flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/RollingSink.java]: Another space before period.

File: flink-connectors/flink-connector-filesystem/src/test/java/org/apache/flink/streaming/connectors/fs/RollingSinkSecuredITCase.java [hunk 13]
```diff
@@ -47,21 +48,21 @@
...
-import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_BLOCK_ACCESS_TOKEN_ENABLE_KEY;
-import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_HTTP_POLICY_KEY;
-import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_ENCRYPT_DATA_TRANSFER_KEY;
-import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_DATANODE_ADDRESS_KEY;
-import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_DATANODE_HOST_NAME_KEY;
 
 /**
  * Tests for running {@link RollingSinkSecuredITCase} which is an extension of {@link RollingSink} in secure environment
- * Note: only executed for Hadoop version > 3.x.x
+ * Note: only executed for Hadoop version > 3.x.x .
```
Comment (greghogan) [inline flink-connectors/flink-connector-filesystem/src/test/java/org/apache/flink/streaming/connectors/fs/RollingSinkSecuredITCase.java]: And another.

File: flink-connectors/flink-avro/src/main/java/org/apache/flink/api/avro/FSDataInputStreamWrapper.java [hunk 14]
```diff
@@ -16,20 +16,19 @@
...
-import org.apache.flink.core.fs.FSDataInputStream;
 
+import java.io.Closeable;
+import java.io.IOException;
 
 /**
- * Code copy pasted from org.apache.avro.mapred.FSInput (which is Apache licensed as well)
- * 
- * The wrapper keeps track of the position in the data stream.
+ * Code copy pasted from org.apache.avro.mapred.FSInput (which is Apache licensed as well.
```
Comment (greghogan) [inline flink-connectors/flink-avro/src/main/java/org/apache/flink/api/avro/FSDataInputStreamWrapper.java]: Lost the closing parenthesis.

**Comment by zentol** [comment]
@greghogan I've fixed all the issues you found. I don't know what to do about the checkstyle disabling though using `//CHECKSTYLE.OFF: ConstantNameCheck|`. The regex evaluation appears to be broken, but maybe it's only me?

File: flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java (hunk 10 above)
Comment (zentol) [inline flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceTestBase.java]: Figured this out, i was supplying the wrong rule to disable, should be `StaticVariableNameCheck` instead.

**Comment by greghogan** [comment]
+1

**Comment by zentol** [comment]
merging.

**Pull request merged**